### PR TITLE
WIP: Render diff tables and font tables

### PR DESCRIPTION
**Pull request by m4rc1e**
For the past year I've had to use GF Regression to visualise diff results.

This PR will allow users to generate their own before and after gifs. In order to do this, Harfbuzz, Cairo and FreeType need to be installed.

I'm hoping this feature will allow designers to easily proof their fonts. Previously it was a pita. This implementation is also over x100 faster.

---
kerns_missing
_Cairo upstream against GF version_

---

@davelab6 keep this open.

I still need to do the following:
- Add installation instructions for the 3rd party dependencies
- Add VF support.
- Add a basic functional test

cc @chrissimpkins

### CODE REVIEW

File: Lib/diffenator/vizualize.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    for cat in diff_dict:
+        if cat not in renderable:
+            continue
+        for sub_cat in diff_dict[cat]:
+            title = '{} {}'.format(cat.title(), sub_cat.title())
+            filename = os.path.join(dst, '{}_{}.gif'.format(cat, sub_cat))
+            diff = diff_dict[cat][sub_cat]
+            if not len(diff) > 1:
+                continue
+            diff_render_table(font_a, font_b, diff[:1000], filename, size, title)
```
Comment (m4rc1e) [inline Lib/diffenator/vizualize.py]: Should not be 1,000. Limit should be the amount of rows in the reports.

**Comment by chrissimpkins** [comment]
`fontdiffenator -rd -r diff.gif [Font1.ttf] [Font2.ttf]` is how you use it?

**Comment by m4rc1e** [comment]
`diffenator font_a font_b -r ~/path/to/dir`

I'll update the Readme.

**Comment by m4rc1e** [comment]
Getting close. I just need to add the C libs to travis.

Once this is merged, I'm make a start on #28

**Comment by m4rc1e** [comment]
@davelab6 This can be merged now.

I haven't extensively tested it so there may be a few edge cases I haven't covered. I'll hopefully pick them up when I refactor to make the v1.0.0 release.

**Comment by m4rc1e** [comment]
@davelab6 we have a bunch of fonts that need releasing this week. I'll test on this first before we merge.

**Comment by davelab6** [comment]
Sounds good

**Comment by m4rc1e** [comment]
@davelab6 we're good here sir!

Apologies for taking a while, had to learn a bunch of new apis and Cython etc.

The functionality of this tool is pretty much complete. I will tackle #28 over the holidays so we have a quality tool .
